### PR TITLE
108 added academic title  and front end validation to registration form

### PR DESCRIPTION
**Pull request by sami1riaz**
- Added Academic Title, First Name, and Last Name to the registration form
- Added Alert element to display backend errors
- Added index.ts to the Components folder so that we do not have to import each component separately

### CODE REVIEW

**Review by redfrexx — COMMENTED** [review]
Thanks! Looks good! 

One question: should we remove the username from the registration process? It seems a bit redundant, since the email is used for authentication later and we require the user to provide a first and last name anyways. Or are there other reasons to keep it?

**Comment by sami1riaz** [comment]
> 

That's a good point. I agree, we should remove it. I will raise an issue for it.

Although a unique ID should still be associated with the user in the backend. If this is not already being done, we should add that functionality in the same issue

**Comment by redfrexx** [comment]
yes sounds good! I can do it tomorrow and push it to this branch.

File: frontend/pages/sign-up.tsx
Comment (codingfabi) [inline frontend/pages/sign-up.tsx:230]: Is there a certain reason or design pattern behind movin the mutations to the bottom of the files?
If not so, I would suggest either leaving them on the top of the files or moving all the mutations to the bottom. I don't care which way we do it, I would just like some consistency :)

File: frontend/pages/sign-up.tsx
Comment (codingfabi) [inline frontend/pages/sign-up.tsx:72]: Great!

File: frontend/pages/sign-in.tsx
Comment (codingfabi) [inline frontend/pages/sign-in.tsx:159]: See comment on other mutation

**Comment by codingfabi** [comment]
Waiting with merge for @redfrexx to push backend changes into this branch

File: frontend/pages/sign-up.tsx
Comment (sami1riaz) [reply frontend/pages/sign-up.tsx:230]: Ah, I just did this so that I wouldn't have to scroll too far down when working on the form. I'll revert this

Comment (sami1riaz) [reply frontend/pages/sign-up.tsx]: done

File: frontend/pages/sign-in.tsx
Comment (sami1riaz) [reply frontend/pages/sign-in.tsx]: done

File: frontend/pages/sign-up.tsx
Comment (codingfabi) [reply frontend/pages/sign-up.tsx]: great

**Comment by codingfabi** [comment]
Looks good to merge in my opinion. Fix the conflicts and feel free to press the merge button :)

**Review by codingfabi — APPROVED** [review]

**Comment by redfrexx** [comment]
> Looks good to merge in my opinion. Fix the conflicts and feel free to press the merge button :)

ah wait, someone still needs to remove the username from the registration form and the queries to the backend before merging. Otherwise it will throw errors I guess.

**Comment by sami1riaz** [comment]
I will remove the username from the frontend and resolve conflicts

**Pull request merged**
